refactor: use slices.Equal to simplify code

minxinyi opened the pull request:
In the Go 1.21 standard library, a new function has been introduced that enhances code conciseness and readability. It can be find  [here](https://pkg.go.dev/slices@go1.21.1#Equal).

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/gohugoio/hugo?pullRequest=13866) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/gohugoio/hugo?pullRequest=13866) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/gohugoio/hugo?pullRequest=13866) it.</sub>

bep commented on navigation/menu_cache.go:
```diff
@@ -48,17 +48,7 @@ type menuCache struct {
-
-	return true
+	return slices.Equal(m1, m2)
 }
```
Thanks for this. This would be even better if you got rid of the menuEqual and compareStringSlices funcs (they're only used in a few places.

dilyanpalauzov commented:
navigation/menu_cache_test.go uses the now deleted `menuEqual()`.

minxinyi commented:
> navigation/menu_cache_test.go uses the now deleted `menuEqual()`.

Oh, Sorry. 

Thank you for your review; I have searched and replaced all instances.
